### DISCUSSION TITLE
Different Responses for same Endpoint

### DISCUSSION THREAD

**Discussion by ercanpinar** [stoplightio/prism#2220] [Q&A]
Hi, 
We use the same endpoint with different request bodies. 
I do not want to make any changes in my code/API layer like "Header - adding Prefer: "example=someExample" ". Is there any other way to get different responses for different request bodies?

**Comment by zergeborg** [comment]
I am in the same boat and I would really find such feature useful. I wish Prism would have a DSL for controlling the preferred behavior INSIDE the OpenAPI doc instead of from the client which initiated the request.

Prism already supports another DSL provided by Faker and JSON schema Faker to allow dynamic response generation, so I don't see any conceptual problem on extending the DSL further.

In fact IMO the current status where Prism takes Prefer header goes completely against using Faker and the idea of declarative programming.
